### PR TITLE
sql/sqlcheck: add rename analyzer

### PR DESCRIPTION
**Pull request by omarhachach**
Converted atlas-contrib implementation to an analyzer.
Tests include only a happy-path test. Follows structure of #853

### CODE REVIEW

File: sql/sqlcheck/sqlcheck_test.go [hunk 1]
```diff
@@ -89,3 +122,35 @@ type testFile struct {
 func (t testFile) Name() string {
 	return t.name
 }
+
+type mockDriver struct {
```
Comment (a8m) [inline sql/sqlcheck/sqlcheck_test.go:126]: The mock does not need to have any logic in it. I should get the desired response as an argument and return it.

File: sql/sqlcheck/sqlcheck.go [hunk 2]
```diff
@@ -133,6 +133,56 @@ var Destructive = &driverAware{
...
+						continue
+					}
+					for j := range c.Changes[i+1:] {
+						a, ok := c.Changes[i+1+j].(*schema.AddColumn)
+						if !ok {
+							continue
+						}
+						// Use schema.Differ to see if type has changed, minimal setup to test only type changes
+						fromTbl := schema.NewTable("f").AddColumns(schema.NewColumn("diff").SetType(d.C.Type.Type))
+						toTbl := schema.NewTable("f").AddColumns(schema.NewColumn("diff").SetType(a.C.Type.Type))
```
Comment (a8m) [inline sql/sqlcheck/sqlcheck.go:158]: The type is not enough.
The columns should look the same. Hence, they need to have the same attributes.

File: sql/sqlcheck/sqlcheck.go [hunk 3]
```diff
@@ -133,6 +133,56 @@ var Destructive = &driverAware{
...
+				if !ok {
+					continue
+				}
+			Loop:
+				for i := range c.Changes {
+					d, ok := c.Changes[i].(*schema.DropColumn)
+					if !ok {
+						continue
+					}
+					for j := range c.Changes[i+1:] {
```
Comment (a8m) [inline sql/sqlcheck/sqlcheck.go:151]: Please, document why we only look forward.
https://github.com/ariga/atlas/blob/master/sql/internal/sqlx/diff.go#L188-L193

**Comment by a8m** [comment]
Something I discussed with @masseelch about it, is that we need to determine (from the statement) if the column was actually dropped and added and not renamed to avoid false-positive reporting. It can be done with regexp for now, but we'll add a proper parser to the statements soon.

**Review by a8m — COMMENTED** [review]
Great! See comments

File: sql/sqlcheck/sqlcheck.go (hunk 2 above)
Comment (masseelch) [reply sql/sqlcheck/sqlcheck.go:158]: True. Also, please make sure to add a test for this.

File: sql/sqlcheck/sqlcheck.go [hunk 4]
```diff
@@ -80,6 +80,14 @@ func (a Analyzers) Analyze(ctx context.Context, p *Pass) error {
 	return nil
 }
 
+// AnalyzeFunc implements Analyzer.
+type AnalyzeFunc func(context.Context, *Pass) error
+
+// Analyze calls a(ctx, p).
+func (a AnalyzeFunc) Analyze(ctx context.Context, p *Pass) error {
+	return a(ctx, p)
```
Comment (a8m) [inline sql/sqlcheck/sqlcheck.go:88]:
```suggestion
func (f AnalyzeFunc) Analyze(ctx context.Context, p *Pass) error {
	return f(ctx, p)
```

File: sql/sqlcheck/sqlcheck.go [hunk 5]
```diff
@@ -133,6 +141,58 @@ var Destructive = &driverAware{
...
+						return err
+					}
+					if len(changes) > 0 {
+						continue
+					}
+					diags = append(diags, Diagnostic{
+						Pos:  sc.Pos,
+						Text: fmt.Sprintf("Potential rename from column %q to %q in table %q", d.C.Name, a.C.Name, c.T.Name),
+					})
+					continue Loop // Be greedy, assume first match is most likely correct.
```
Comment (a8m) [inline sql/sqlcheck/sqlcheck.go:182]: Why not break instead?

File: sql/sqlcheck/sqlcheck.go [hunk 6]
```diff
@@ -133,6 +141,58 @@ var Destructive = &driverAware{
 	},
 }
 
+// Renames detects potential column renames.
+var Renames = AnalyzeFunc(func(ctx context.Context, p *Pass) error {
+	diags := make([]Diagnostic, 0)
```
Comment (a8m) [inline sql/sqlcheck/sqlcheck.go:146]:
```suggestion
	var diags []Diagnostic
```

**Pull request closed without merging**